test/integration/framework/etcd.go

```diff
@@ -128,7 +129,18 @@ func RunCustomEtcd(dataDir string, customFlags []string) (url string, stopFn fun
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	stop := func() {
-		cancel()
+		// try to exit etcd gracefully
+		defer cancel()
+		cmd.Process.Signal(syscall.SIGTERM)
+		go func() {
+			select {
+			case <-ctx.Done():
+				klog.Infof("etcd exited gracefully, context cancelled")
```

Comment:
waiting on `ctx.Done()` here doesn't mean the etcd process exited gracefully, it means our `cancel` was called...

Reply:
The key here is the cmd.Wait(), that is blocking until the process exits, if it exitss gracefully the `cancel` on the defer is executed , it only happens if the process exits because of the SIGTERM in less than 5 seconds ... the other statement in this select is the one that forcefully exits the process executing the cancel()

Reply:
oh, I see. ok